main.cpp

```diff
@@ -54,14 +56,21 @@ int main(int argc, char** argv)
     // default logger
     Log = new STDIOHandler();
 
-    Configuration* configuration = new Configuration();
-    // Parsing the arguments
-    bool status = false;
-    {
-          //ArgumentParser* p = new ArgumentParser(configuration); //Memory Leak
-          std::unique_ptr<ArgumentParser> p(new ArgumentParser(configuration));
-          status = p->ParseArgs(argc, argv);
+    ArgumentParser a;
+    if(auto [execution, return_code] = a.Parse(argc, argv);
+       execution == ArgumentParser::Execution::ABORT) {
+        return return_code;
     }
+
+    Configuration config;
+
+    //TODO remove me in refactoring
+    Configuration* configuration = &config;
```

Comment:
Whats the problem with removing this pointer? Is it used somewhere outside this file?